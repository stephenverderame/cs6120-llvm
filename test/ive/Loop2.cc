#include <cstdio>
int main()
{
    for (int I = 10; I >= 0; --I) {
        printf("%d\n", I);
    }
}