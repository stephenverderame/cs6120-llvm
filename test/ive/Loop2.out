Loop: 
1:                                                ; preds = %6, %0
  %.0 = phi i32 [ 10, %0 ], [ %7, %6 ]
  %2 = icmp sge i32 %.0, 0
  br i1 %2, label %4, label %3

Basic IV:   %.0 = phi i32 [ 10, %0 ], [ %7, %6 ]
Basic IV:   %7 = add nsw i32 %.0, -1
